crates/ty_python_semantic/src/types.rs

```diff
@@ -8046,6 +8048,105 @@ impl<'db> CallableType<'db> {
+                fn any(db: &dyn Db) -> Type<'_> {
+                    CallableType::single(
+                        db,
+                        Signature::new(Parameters::unknown(), Some(Type::any())),
+                    )
+                }
+
+                any(db)
+            }
+            DynamicType::Unknown => {
+                #[salsa::tracked]
+                fn unknown(db: &dyn Db) -> Type<'_> {
+                    CallableType::single(
+                        db,
+                        Signature::new(Parameters::unknown(), Some(Type::unknown())),
+                    )
+                }
+
+                unknown(db)
+            }
+            #[cfg(debug_assertions)]
+            DynamicType::Todo(_) => CallableType::single(
+                db,
+                Signature::new(Parameters::unknown(), Some(Type::Dynamic(return_type))),
+            ),
+            // The TodoType is a singleton in release builds
+            #[cfg(not(debug_assertions))]
+            DynamicType::Todo(_) => {
+                #[salsa::tracked]
+                fn todo<'db>(db: &'db dyn Db) -> Type<'db> {
+                    CallableType::single(
+                        db,
+                        Signature::new(
+                            Parameters::unknown(),
+                            Some(Type::Dynamic(DynamicType::Todo(TodoType))),
+                        ),
+                    )
+                }
+
+                todo(db)
+            }
+            DynamicType::TodoPEP695ParamSpec => {
+                #[salsa::tracked]
+                fn pep695(db: &dyn Db) -> Type<'_> {
+                    CallableType::single(
+                        db,
+                        Signature::new(
+                            Parameters::unknown(),
+                            Some(Type::Dynamic(DynamicType::TodoPEP695ParamSpec)),
+                        ),
+                    )
+                }
+
+                pep695(db)
+            }
+            DynamicType::TodoTypeAlias => {
+                #[salsa::tracked]
+                fn type_alias(db: &dyn Db) -> Type<'_> {
+                    CallableType::single(
+                        db,
+                        Signature::new(
+                            Parameters::unknown(),
+                            Some(Type::Dynamic(DynamicType::TodoTypeAlias)),
+                        ),
+                    )
+                }
+
+                type_alias(db)
+            }
+            DynamicType::TodoUnpack => {
+                #[salsa::tracked]
+                fn unpack(db: &dyn Db) -> Type<'_> {
+                    CallableType::single(
+                        db,
+                        Signature::new(
+                            Parameters::unknown(),
+                            Some(Type::Dynamic(DynamicType::TodoUnpack)),
+                        ),
+                    )
+                }
+                unpack(db)
+            }
+        }
+    }
```

Comment:
I think these should all use `Parameters::gradual_form()` rather than `Parameters::unknown()`, if we want to keep the semantics exactly as they are on `main`. The former creates a `*args: Any, **kwargs: Any` parameter spec, the second creates a `*args: Unknown, **kwargs: Unknown` parameter spec. Practically speaking these are usually treated the same by ty, but `Unknown` normalizes to `Any`, and it could also have an effect on diagnostics messages.